message/validation/peer_limiter.go

```diff
@@ -0,0 +1,101 @@
+package validation
+
+import (
+	"sync"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/time/rate"
+)
+
+type RateLimiter struct {
+	rejectLimiter *rate.Limiter
+	ignoreLimiter *rate.Limiter
+	blockRequests bool
+	lastBlocked   time.Time
+	mu            sync.Mutex // Mutex to protect blockRequests and lastBlocked
+}
+
+func NewRateLimiter(rejectLimit, ignoreLimit rate.Limit) *RateLimiter {
+	return &RateLimiter{
+		rejectLimiter: rate.NewLimiter(rejectLimit, int(rejectLimit)),
```

Comment:
maybe like you said in our conversation
if this is just for RSA sig errors we should give this a less confusing name